rasterio/vrt.py

```diff
@@ -244,4 +244,58 @@ def _boundless_vrt_doc(
         dstrect.attrib['xSize'] = str(src_dataset.width)
         dstrect.attrib['ySize'] = str(src_dataset.height)
 
-    return ET.tostring(vrtdataset).decode('ascii')
+    return ET.tostring(vrtdataset, encoding="utf-8")
+
+
+class VirtualDataset:
+    """A lazily evaluated virtual dataset based on GDAL's VRT."""
+
+    @classmethod
+    def fromstring(cls, text):
```

Comment:
Do you foresee adding methods of initialization of a VirtualDataset?

Reply:
@snowman2 yes, I think so. The VirtualDataset constructor could easily take the profile keyword arguments (height, width, transform, crs, etc). That would be a good start, do you think? Bands would be difficult, there are so many more parameters involved. I think we'd need to rely on an addBand type of method.

Reply:
It occurs to me that most of addBand's signature looks like dataset.read's. All except the `<DstRect>` element, yeah?

Reply:
Do you think there would be value in a VRTBand class that you could pass into the constructor?